diffusion/models/stable_diffusion.py

```diff
@@ -448,14 +452,24 @@ def generate(
         added_cond_kwargs = {}
         # if using SDXL, prepare added time ids & embeddings
         if self.sdxl and pooled_embeddings is not None:
-            if not crop_params:
-                crop_params = [0., 0.]
-            if not size_params:
-                size_params = [width, height]
-            add_time_ids = torch.tensor([[width, height, *crop_params, *size_params]], dtype=torch.float, device=device)
-            add_time_ids = add_time_ids.repeat(pooled_embeddings.shape[0], 1)
-            add_text_embeds = pooled_embeddings
+            if crop_params is None:
+                crop_params = torch.tensor([0., 0.]).repeat(batch_size, 1)
+            elif isinstance(crop_params, tuple):  # convert to tensor
+                crop_params = torch.tensor([crop_params[0], crop_params[1]]).repeat(batch_size, 1).float()
+            if input_size_params is None:
+                input_size_params = torch.tensor([width, height]).repeat(batch_size, 1).float()
+            elif isinstance(input_size_params, tuple):  # convert to tensor
+                input_size_params = torch.tensor([input_size_params[0], input_size_params[1]]).repeat(batch_size,
+                                                                                                      1).float()
+            output_size_params = torch.tensor([width, height]).repeat(batch_size, 1).float()
```

Comment:
Just construct the tensor with dtype float?

Reply:
done!